src/hotspot/share/opto/compile.cpp

```diff
@@ -611,77 +611,75 @@ void Compile::print_ideal_ir(const char* phase_name) {
-                  _node_arena_two(mtCompiler, Arena::Tag::tag_node),
-                  _node_arena(&_node_arena_one),
-                  _mach_constant_base_node(nullptr),
-                  _Compile_types(mtCompiler),
-                  _initial_gvn(nullptr),
-                  _igvn_worklist(nullptr),
-                  _types(nullptr),
-                  _node_hash(nullptr),
-                  _late_inlines(comp_arena(), 2, 0, nullptr),
-                  _string_late_inlines(comp_arena(), 2, 0, nullptr),
-                  _boxing_late_inlines(comp_arena(), 2, 0, nullptr),
-                  _vector_reboxing_late_inlines(comp_arena(), 2, 0, nullptr),
-                  _late_inlines_pos(0),
-                  _number_of_mh_late_inlines(0),
-                  _oom(false),
-                  _print_inlining_stream(new (mtCompiler) stringStream()),
-                  _print_inlining_list(nullptr),
-                  _print_inlining_idx(0),
-                  _print_inlining_output(nullptr),
-                  _replay_inline_data(nullptr),
-                  _java_calls(0),
-                  _inner_loops(0),
-                  _interpreter_frame_size(0),
-                  _output(nullptr)
+      _has_method_handle_invokes(false),
+      _clinit_barrier_on_entry(false),
+      _stress_seed(0),
+      _comp_arena(mtCompiler),
+      _barrier_set_state(BarrierSet::barrier_set()->barrier_set_c2()->create_barrier_state(comp_arena())),
+      _env(ci_env),
+      _directive(directive),
+      _log(ci_env->log()),
+      _first_failure_details(nullptr),
+      _intrinsics(comp_arena(), 0, 0, nullptr),
+      _macro_nodes(comp_arena(), 8, 0, nullptr),
+      _parse_predicates(comp_arena(), 8, 0, nullptr),
+      _template_assertion_predicate_opaqs(comp_arena(), 8, 0, nullptr),
+      _expensive_nodes(comp_arena(), 8, 0, nullptr),
+      _for_post_loop_igvn(comp_arena(), 8, 0, nullptr),
+      _unstable_if_traps(comp_arena(), 8, 0, nullptr),
+      _coarsened_locks(comp_arena(), 8, 0, nullptr),
+      _congraph(nullptr),
+      NOT_PRODUCT(_igv_printer(nullptr) COMMA)
+          _unique(0),
+      _dead_node_count(0),
+      _dead_node_list(comp_arena()),
+      _node_arena_one(mtCompiler, Arena::Tag::tag_node),
+      _node_arena_two(mtCompiler, Arena::Tag::tag_node),
+      _node_arena(&_node_arena_one),
+      _mach_constant_base_node(nullptr),
+      _Compile_types(mtCompiler),
+      _initial_gvn(nullptr),
+      _igvn_worklist(nullptr),
+      _types(nullptr),
+      _node_hash(nullptr),
+      _late_inlines(comp_arena(), 2, 0, nullptr),
+      _string_late_inlines(comp_arena(), 2, 0, nullptr),
+      _boxing_late_inlines(comp_arena(), 2, 0, nullptr),
+      _vector_reboxing_late_inlines(comp_arena(), 2, 0, nullptr),
+      _late_inlines_pos(0),
+      _number_of_mh_late_inlines(0),
+      _oom(false),
+      _replay_inline_data(nullptr),
+      _inline_printer(comp_arena(), directive->PrintInliningOption || PrintOptoInlining),
```

Comment:
How do we support print_intrinsics()?

Reply:
That's a good point. Inline printer should be enabled if C->print_intrinsics() or C->print_inlining() is true so this change does seem to change the behavior. I'll fix this